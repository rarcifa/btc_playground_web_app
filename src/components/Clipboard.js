import * as React from 'react';
import { InputAdornment } from '@material-ui/core';
import IconButton from '@material-ui/core/IconButton';
import FileCopyOutlinedIcon from '@material-ui/icons/FileCopyOutlined';

function Clipboard() {

    return (
        <InputAdornment position="start">
            <IconButton 
            aria-label="toggle password visibility"
            onClick={() => {navigator.clipboard.writeText(this.state.textToCopy)}}>
                <FileCopyOutlinedIcon />
            </IconButton>
        </InputAdornment>
    );
}

export default Clipboard;