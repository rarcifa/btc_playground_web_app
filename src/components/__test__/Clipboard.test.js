import React from 'react';
import { mount, shallow } from 'enzyme';
import { IconButton, InputAdornment, MuiThemeProvider } from '@material-ui/core';
import { FileCopyOutlinedIcon } from '@material-ui/icons';
import Clipboard from './../Clipboard';
import theme from './../../theme';

const wrapper = shallow(
  <MuiThemeProvider theme={theme}>
    <Clipboard/>
  </MuiThemeProvider>
)

describe('Clipboard', () => {
  it('should render the Clipboard Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});