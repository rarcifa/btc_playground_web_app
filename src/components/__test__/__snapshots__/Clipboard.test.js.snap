// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clipboard should render the Clipboard Component correctly 1`] = `ShallowWrapper {}`;
